src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -167,6 +170,11 @@ def __runner():
                             RobotStatus, robot_status
                         )
 
+                        # blocking wait for primktivies to keep world in sync with them
+                        self.primitive_set_buffer.get(
+                            block=True, timeout=WORLD_BUFFER_TIMEOUT
+                        )
```

Comment:
it's weird that this blocking wait is happening in the exception, doesn't seem like it'd affect the normal code path